### PR TITLE
Move Course Edit Page over to the new saving system

### PR DESCRIPTION
**Pull request by dmcavoy**
A while back we moved Lesson and Script edit pages over to using a saving system that allowed an editor to save and close or saving and keep editing. This system also allows us more control over saving and preventing saving in certain situations. We did not move the Course Edit page at that time. When I was working on the work to update the way we publish courses (https://github.com/code-dot-org/code-dot-org/pull/40537) I realized it would be very helpful to have this working on the course edit page so I quickly set it up.

Screen Shot 2021-05-17 at 10 21 02 AM

## Links

Did in prep for work of: https://codedotorg.atlassian.net/browse/PLAT-1050

## Testing story

Updated lots of existing unit tests and added some new tests for saving in the CourseEditor.
Also manually tried it locally.

### CODE REVIEW

File: apps/src/lib/levelbuilder/course-editor/CourseEditor.js [hunk 1]
```diff
@@ -57,44 +60,107 @@ class CourseEditor extends Component {
...
         <h1>{name}</h1>
         <label>
           Display Name
-          <input
-            type="text"
-            name="title"
-            defaultValue={title}
-            style={styles.input}
-          />
+          <input type="text" defaultValue={title} style={styles.input} />
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: I think you might need to have an `onChange` handler here. When I update the display name locally, the change doesn't stick

Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: Actually, this seems true for a few fields here (also tested short description, version year, and unit numbering)

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: Oh good catch. I did not add that. Thanks for your thoughtful review

File: apps/src/lib/levelbuilder/course-editor/CourseEditor.js [hunk 2]
```diff
@@ -197,11 +268,7 @@ class CourseEditor extends Component {
         <CollapsibleEditorSection title="Publishing Settings">
           <label>
             Family Name
-            <select
-              name="family_name"
-              defaultValue={familyName}
-              style={styles.dropdown}
-            >
+            <select defaultValue={familyName} style={styles.dropdown}>
```
Comment (bethanyaconnor) [inline apps/src/lib/levelbuilder/course-editor/CourseEditor.js]: I think you also need `onChange` for `familyName` and `versionYear`

**Review by bethanyaconnor — APPROVED** [review]
Yay I'm excited for this -- it'll make developing on the course edit page so much easier! I noticed a couple more inputs that need to be updated but otherwise looks great!

**Pull request merged**